artifactory/utils/dependenciesutils.go

```diff
@@ -39,25 +40,40 @@ func DownloadExtractorIfNeeded(targetPath, downloadPath string) error {
 }
 
 func GetExtractorsRemoteDetails(downloadPath string) (*config.ServerDetails, string, error) {
+	releasesRemote := os.Getenv(releasesRemoteEnv)
+	if releasesRemote != "" {
+		return getRemoteDetails(releasesRemote, downloadPath, releasesRemoteEnv)
+	}
+
+	// Fallback for the deprecated JFROG_CLI_EXTRACTORS_REMOTE environment variable
 	extractorsRemote := os.Getenv(ExtractorsRemoteEnv)
 	if extractorsRemote != "" {
-		return getExtractorsRemoteDetails(extractorsRemote, downloadPath)
+		return getRemoteDetails(extractorsRemote, downloadPath, ExtractorsRemoteEnv)
 	}
+
 	log.Info("The build-info-extractor jar is not cached locally. Downloading it now...\nYou can set the repository from which this jar is downloaded. Read more about it at https://www.jfrog.com/confluence/display/CLI/CLI+for+JFrog+Artifactory#CLIforJFrogArtifactory-DownloadingtheMavenandGradleExtractorJARs")
-	log.Debug("'" + ExtractorsRemoteEnv + "' environment variable is not configured. Downloading directly from releases.jfrog.io.")
+	log.Debug("'" + releasesRemoteEnv + "' environment variable is not configured. Downloading directly from releases.jfrog.io.")
 	// If not configured to download through a remote repository in Artifactory, download from releases.jfrog.io.
 	return &config.ServerDetails{ArtifactoryUrl: "https://releases.jfrog.io/artifactory/"}, path.Join("oss-release-local", downloadPath), nil
 }
 
-// Get Artifactory server details and a repository proxying oss.jfrog.org according to JFROG_CLI_EXTRACTORS_REMOTE env var.
-func getExtractorsRemoteDetails(extractorsRemote, downloadPath string) (*config.ServerDetails, string, error) {
-	lastSlashIndex := strings.LastIndex(extractorsRemote, "/")
+// For JFROG_CLI_EXTRACTORS_REMOTE: get Artifactory server details and a repository proxying oss.jfrog.org according to JFROG_CLI_EXTRACTORS_REMOTE env var.
+// For JFROG_CLI_RELEASES_REPO: get Artifactory server details and a repository proxying https://releases.jfrog.io according to JFROG_CLI_RELEASES_REPO env var.
```

Comment:
The above function description isn't clear for the random reader. It starts with "For JFROG_CLI_EXTRACTORS_REMOTE: ...", instead of a general description of rhe function. The details should definately follow the general description. Always try reading the description while wearing a hat of a user with very little context.
Let's also describe the arguments received by this function.

Reply:
done